pkgs/development/python-modules/markdownsuperscript/default.nix

```diff
@@ -0,0 +1,22 @@
+{ stdenv, buildPythonPackage, fetchPypi, markdown }:
+
+buildPythonPackage rec {
+  pname = "MarkdownSuperscript";
+  version = "2.0.0";
+  name = "${pname}-${version}";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1dsx21h9hkx098d5azpw81dcz23rrgzrwlymwv7jri348q26p748";
+  };
+
+  propagatedBuildInputs = [ markdown ];
+
+  doCheck = false; # Tests depend on packages which are not available yet in nixpkgs
```

Comment:
@FRidh Today, I've tried activating the tests for this package. First, I've added the missing dependency (see #27018). Then, I've made the following changes to this derivation so that it would run the tests:

```diff
diff --git a/pkgs/development/python-modules/markdownsuperscript/default.nix b/pkgs/development/python-modules/markdownsuperscript/default.nix
index fdf9913e76..9677de0c99 100644
--- a/pkgs/development/python-modules/markdownsuperscript/default.nix
+++ b/pkgs/development/python-modules/markdownsuperscript/default.nix
@@ -1,18 +1,24 @@
-{ stdenv, buildPythonPackage, fetchPypi, markdown }:
+{ stdenv, buildPythonPackage, fetchFromGitHub, markdown, nose, codecov }:
 
 buildPythonPackage rec {
   pname = "MarkdownSuperscript";
   version = "2.0.0";
   name = "${pname}-${version}";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "1dsx21h9hkx098d5azpw81dcz23rrgzrwlymwv7jri348q26p748";
+  src = fetchFromGitHub {
+    owner = "jambonrose";
+    repo = "markdown_superscript_extension";
+    rev = "v${version}";
+    sha256 = "0kvnvicgnxmd9xlm9gi7nk9jqnhj2qp7238xxrxls385l0x07004";
   };
 
+  buildInputs = [ nose codecov ]; # Tests only
+
   propagatedBuildInputs = [ markdown ];
 
-  doCheck = false; # Tests depend on packages which are not available yet in nixpkgs
+  patchPhase = ''
+    sed -i 's/==.*$//' requirements/test_requirements.txt
+  '';
 
   meta = {
     description = "An extension to the Python Markdown package enabling superscript text";
```

You may remark in passing that I've changed the source from PyPi to GitHub because initially I got the following error: `ImportError: No module named tests`. And indeed, checking the source from PyPi I could see that the `tests/` folder which is present in the source on GitHub wasn't there. However, after this change the tests still fail in the same way in Python 2 and in Python 3, I get a new error (showing only the tail of the trace):

```
  File "/nix/store/4irqw5y5mj9dr41x1fl4cgqxpyaqvlpj-python3.6-bootstrapped-pip-9.0.1/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1521, in _listdir
    "Can't perform this operation for unregistered loader type"
NotImplementedError: Can't perform this operation for unregistered loader type
```

I don't know if this is a problem of version of a dependency or something else.
I'm also wondering if these tests are meant to be run by the user (there are no instructions on how to run them).
I'm not a Python programmer and I feel like I've lost already way too much time on this. Feel free to take over, to merge or to close the PR.